lib/ReactViews/ExplorerWindow/Tabs/MyDataTab/CesiumIonConnector.tsx

```diff
@@ -0,0 +1,540 @@
+  const [assets, setAssets] = React.useState<CesiumIonAsset[]>([]);
+  const [isLoadingAssets, setIsLoadingAssets] = React.useState<boolean>(false);
+
+  const [loginToken, setLoginToken] = React.useState(
+    localStorage.getItem(tokenLocalStorageName) ?? ""
+  );
+
+  const defaultProfile = {
+    id: 0,
+    scopes: [],
+    username: "",
+    email: "",
+    emailVerified: false,
+    avatar: string,
+    storage: {}
+  };
+  const [userProfile, setUserProfile] = React.useState(defaultProfile);
+  const [isLoadingUserProfile, setIsLoadingUserProfile] =
+    React.useState<boolean>(false);
+
+  React.useEffect(() => {
+    if (!crypto || !crypto.subtle) return;
+
+    const codeChallenge = [...crypto.getRandomValues(new Uint8Array(32))]
+      .map((x) => x.toString(16).padStart(2, "0"))
+      .join("");
+
+    crypto.subtle
+      .digest("SHA-256", new TextEncoder().encode(codeChallenge))
+      .then((hash) => {
+        setCodeChallenge({
+          value: codeChallenge,
+          hash: btoa(String.fromCharCode(...new Uint8Array(hash)))
+            .replace(/\+/g, "-")
+            .replace(/\//g, "_")
+            .replace(/[=]/g, "")
+        });
+      });
+  }, []);
+
+  React.useEffect(() => {
+    if (loginToken.length === 0) return;
+
+    setIsLoadingUserProfile(true);
+
+    fetch("https://api.cesium.com/v1/me", {
+      headers: {
+        Authorization: `Bearer ${loginToken}`
+      }
+    })
+      .then((response) => {
+        return response.json();
+      })
+      .then((profile) => {
+        if (profile && profile.username) {
+          setUserProfile(profile);
+        } else {
+          setUserProfile(defaultProfile);
+        }
+        setIsLoadingUserProfile(false);
+      });
+  }, [loginToken]);
+
+  React.useEffect(() => {
+    if (loginToken.length === 0) return;
+
+    setIsLoadingAssets(true);
+
+    fetch("https://api.cesium.com/v1/assets", {
+      headers: {
+        Authorization: `Bearer ${loginToken}`
+      }
+    })
+      .then((response) => response.json())
+      .then((assets: { items: CesiumIonAsset[] }) => {
+        if (assets.items) {
+          assets.items.forEach((item) => {
+            item.uniqueName = `${item.name} (${item.id})`;
+          });
+          setAssets(assets.items);
+        }
+
+        setIsLoadingAssets(false);
+      });
+  }, [loginToken]);
+
+  React.useEffect(() => {
+    if (loginToken.length === 0) return;
+
+    setIsLoadingTokens(true);
+
+    fetch("https://api.cesium.com/v2/tokens", {
+      headers: {
+        Authorization: `Bearer ${loginToken}`
+      }
+    })
+      .then((response) => response.json())
+      .then((tokens: { items: CesiumIonToken[] }) => {
+        if (tokens.items) {
+          tokens.items.forEach((item) => {
+            item.uniqueName = `${item.name} (${item.id})`;
+          });
+          setTokens(tokens.items);
+        }
+        setIsLoadingTokens(false);
+      });
+  }, [loginToken]);
```

Comment:
The login token _is_ a Cesium ion token. It can be used to access assets, for example. But it was obtained by logging in via OAuth2. I think `loginToken` is a little bit better name than `oAuth2Token` or similar, but I added some comments to clarify.